File: core/src/main/java/org/ligi/vaporizercontrol/VaporizerDataBinder.java
package org.ligi.vaporizercontrol;

import android.app.Activity;
import android.widget.TextView;

import org.ligi.vaporizercontrol.model.Settings;
import org.ligi.vaporizercontrol.model.VaporizerData;
import org.ligi.vaporizercontrol.util.TemperatureFormatter;

import ligi.org.core.R;

public class VaporizerDataBinder {

    TextView battery;
    TextView temperature;
    TextView temperatureSetPoint;
    TextView tempBoost;
    TextView led;

    private final Settings settings;

    public VaporizerDataBinder(final Activity context, Settings settings) {
        this.settings = settings;
        battery = (TextView) context.findViewById(R.id.battery);
        temperature = (TextView) context.findViewById(R.id.temperature);
        temperatureSetPoint = (TextView) context.findViewById(R.id.temperatureSetPoint);
        tempBoost = (TextView) context.findViewById(R.id.tempBoost);
        led = (TextView) context.findViewById(R.id.led);
    }

    public void bind(VaporizerData data) {

        battery.setText((data.batteryPercentage == null ? "?" : "" + data.batteryPercentage) + "%");

        temperature.setText(TemperatureFormatter.INSTANCE.getFormattedTemp(settings, data.currentTemperature, true) + " / ");
        temperatureSetPoint.setText(TemperatureFormatter.INSTANCE.getFormattedTemp(settings, data.setTemperature, true));
        tempBoost.setText("+" + TemperatureFormatter.INSTANCE.getFormattedTemp(settings, data.boostTemperature, false));
        led.setText((data.ledPercentage == null ? "?" : "" + data.ledPercentage) + "%");

    }
}

File: core/src/main/java/org/ligi/vaporizercontrol/VaporizerDataBinder.kt
package org.ligi.vaporizercontrol

import android.app.Activity
import android.view.View
import kotlinx.android.synthetic.main.values_layout.*
import org.ligi.vaporizercontrol.model.Settings
import org.ligi.vaporizercontrol.model.VaporizerData
import org.ligi.vaporizercontrol.util.TemperatureFormatter.getFormattedTemp

class VaporizerDataBinder(val context: Activity, private val settings: Settings) {

    fun bind(data: VaporizerData) {
        context.battery.text = if (data.batteryPercentage == null) "?" else "" + data.batteryPercentage!! + "%"

        context.temperature.text = getFormattedTemp(settings, data.currentTemperature, true) + " / "
        context.temperatureSetPoint.text = getFormattedTemp(settings, data.setTemperature, true)
        context.tempBoost.text = "+" + getFormattedTemp(settings, data.boostTemperature, false)
        context.led.text = (if (data.ledPercentage == null) "?" else "" + data.ledPercentage!!) + "%"

        val isNotSeenForSomeTime = System.currentTimeMillis().minus(data.lastDataMillis ?: 0) > 1000
        context.visibility_indicator.visibility = if (isNotSeenForSomeTime) View.VISIBLE else View.INVISIBLE
    }
}
